From 93f120688833bdf9c5ee17c1ad5d26c6b0ff8f50 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 13 Jun 2014 00:33:24 +0100
Subject: [PATCH 2/2] Revert "Revert "Revert "MSYS2 has mounted /usr to / and
 we need remove usr/ prefix from extracted files because it break installing
 to custom root."""

This reverts commit 390d2efe62a53bc4abe365f88ca5689483bfabff.
---
 lib/libalpm/add.c  | 6 +-----
 lib/libalpm/util.c | 5 -----
 2 files changed, 1 insertion(+), 10 deletions(-)

diff --git a/lib/libalpm/add.c b/lib/libalpm/add.c
index 1c6e031..2bcf537 100644
--- a/lib/libalpm/add.c
+++ b/lib/libalpm/add.c
@@ -186,11 +186,7 @@ static int extract_single_file(alpm_handle_t *handle, struct archive *archive,
 			return 0;
 		}
 		/* build the new entryname relative to handle->root */
-		if (strncmp(entryname, "usr/", 4) == 0) {
-			snprintf(filename, PATH_MAX, "%s%s", handle->root, entryname+4);
-		} else {
-			snprintf(filename, PATH_MAX, "%s%s", handle->root, entryname);
-		}
+		snprintf(filename, PATH_MAX, "%s%s", handle->root, entryname);
 	}
 
 	/* if a file is in NoExtract then we never extract it */
diff --git a/lib/libalpm/util.c b/lib/libalpm/util.c
index be12fb7..0b6e714 100644
--- a/lib/libalpm/util.c
+++ b/lib/libalpm/util.c
@@ -373,11 +373,6 @@ int _alpm_unpack(alpm_handle_t *handle, const char *path, const char *prefix,
 		}
 
 		/* Extract the archive entry. */
-		if (strncmp(entryname, "usr/", 4) == 0) {
-			char filename[PATH_MAX];
-			snprintf(filename, PATH_MAX, "%s", entryname+4);
-			archive_entry_set_pathname(entry, filename);
-		}
 		int readret = archive_read_extract(archive, entry, 0);
 		if(readret == ARCHIVE_WARN) {
 			/* operation succeeded but a non-critical error was encountered */
-- 
2.0.0

